azure/target/blobstorage/blobStorageFunction/index.js

```diff
@@ -0,0 +1,244 @@
+/*
+Copyright 2018 VMware, Inc.
+SPDX-License-Identifier: MIT
+*/
+
+/**
+ * Constants
+ */
+const {
+    LIntHttpEnv,
+    Collector,
+    flattenJson,
+    shortenKey,
+} = require('./lint');
+
+/**
+ * This is the base method that will act as entrypoint for the azure supported trigger.
+ * @param context
+ * @param eventHubMessages
+ * @returns {Promise<void>}
+ */
+module.exports = async function (context, eventHubMessages) {
+    handler(eventHubMessages, context);
+};
+
+/**
+ * Converts raw json string to json object for blob.
+ * @param logText
+ * @returns {{}}
+ */
+const processLogTextAsJson = (logText) => {
+    const keys = [];
+    const mergedRecords = {};
+    //mergedRecords.blob = logText;
+    try {
+        let textJson = JSON.parse(logText);
+        textJson = flattenJson(textJson);
+        if ((textJson.timestamp) &&
+            (typeof textJson.timestamp === 'string')) {
+            const numericTimestamp = parseInt(textJson.timestamp, 10);
+            textJson.timestamp = numericTimestamp || textJson.timestamp;
+        }
+        Object.keys(textJson).forEach((key) => {
+            let value = textJson[key];
+            if (value != null) {
+                key = shortenKey(key);
+                if (typeof keys[key] !== 'undefined') {
+                    value = `${mergedRecords[key]} ${value}`;
+                }
+                keys.push(key);
+            }
+            mergedRecords[key] = value;
+        });
+        return mergedRecords;
+    } catch (e) {
+        return mergedRecords;
+    }
+};
+
+/**
+ * Processing logs by adding ingest_timestamp,log_type to log json object
+ * @param LogRecords
+ */
+const processLogs = (LogRecords, logSource) => {
+    const ingestionTime = Date.now();
+    if (typeof LogRecords === 'object') {
+        if (LogRecords !== undefined) {
+            fetchServiceName(LogRecords);
+            LogRecords.ingest_timestamp = ingestionTime;
+            LogRecords.log_type = 'azure_log';
+            LogRecords.logsource = logSource;
+        } else {
+            LogRecords.forEach((record) => {
+                fetchServiceName(record);
+                record.ingest_timestamp = ingestionTime;
+                record.log_type = 'azure_log';
+                record.logsource = logSource;
+            });
+        }
+    }
+};
+
+/**
+ * Fetch service name and provider from the resource ID string
+ * and adding those properties to event_provider,eventsource.
```

Comment:
please add an example of how resourceID will look like and the end result on what this method. does with that

Reply:
Added example resourceId and result